src/frontend/localize_phase.nit

```diff
@@ -0,0 +1,122 @@
+# Basic support of internationalization through the generation of id-to-string tables
+module localize_phase
+
+intrude import literal
+private import annotation
+private import parser_util
+import astbuilder
+import gettext
+
+redef class ToolContext
+	# Main phase of `language`
+	var localize_phase: Phase = new LocalizePhase(self, [literal_phase])
+end
+
+private class LocalizePhase
+	super Phase
+
+	redef fun process_annotated_node(nmodule, nat) do
+		if not nat.name == "localized" then return
+
+		if not nmodule isa AModuledecl then
+			toolcontext.error(nmodule.location, "Error: The localized language can only be used on module declarations.")
+			return
+		end
+
+		var domain = nmodule.n_name.n_id.text
+
+		var haslang = false
+		var lang = ""
+		if nat.n_args.length == 1 then
+			if not nat.n_args[0] isa AStringFormExpr then
+				toolcontext.error(nat.location, "Error: expected string argument for annotation language")
+				return
+			end
+			haslang = true
+			lang = nat.n_args[0].as(AStringFormExpr).value.as(not null)
```

Comment:
Use the existing function

~~~nit
if nat.n_args.length > 0 then
   var lang = nat.arg_as_string # error checking and string extraction
   if lang == null then return # forwar error
~~~